test/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasonsSpec.scala

```diff
@@ -77,6 +77,21 @@ class IabdTaxCodeChangeReasonsSpec extends PlaySpec
       )
     }
 
+    "return all new allowances/deductions in the correct order when you have multiple new allowances/deductions" in {
+      val newDebt = CodingComponentPair(EstimatedTaxYouOweThisYear, None, None, Some(123))
+      val newAllowance = CodingComponentPair(JobExpenses, None, None, Some(123))
+      val newDeduction = CodingComponentPair(CarBenefit, None, None, Some(123))
+
+      val pairs = AllowancesAndDeductionPairs(Seq(newAllowance), Seq(newDebt, newDeduction))
+
+      val reasons = iabdTaxCodeChangeReasons.reasons(pairs)
+
+      reasons mustBe Seq(
+        "You now get Job expenses worth £123",
+        "You now get Car benefit worth £123",
+        "We estimate you have underpaid £24 tax this year")
+    }
+
```

Comment:
Missing tests covering ammended benefits, not sure where they actually go in the new ordering?

Reply:
I've updated it now to test all iabd ordering